### PR TITLE
feat: allow position modification in k8s [DET-6967] [DET-6968]

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
Adds previously available features of the job queue into Kubernetes. Most of the changes are similar to the changes made to the agent job queue.

## Test Plan
Play with the job queue in kubernetes. Submit both single, distributed, and adaptive jobs and rearrange their priorities and positions.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by hamidzr — COMMENTED** [review]
review still in progress : ). I'm trying out the cluster.

File: master/internal/resourcemanagers/resource_pool.go [hunk 1]
```diff
@@ -395,13 +395,11 @@ func (rp *ResourcePool) receiveAgentMsg(ctx *actor.Context) error {
 }
 
 func (rp *ResourcePool) moveJob(
-	ctx *actor.Context, jobID model.JobID, anchorID model.JobID, aheadOf bool,
+	ctx *actor.Context,
+	jobID model.JobID,
+	anchorID model.JobID,
+	aheadOf bool,
 ) error {
-	if rp.config.Scheduler.GetType() != config.PriorityScheduling {
```
[the remark this answers is not in the document]
Comment (eecsliu) [reply master/internal/resourcemanagers/resource_pool.go:419]: ah good catch - I was attempting to make this function more generic, but it turned out to be not really worth the effort. I forgot to restore this part.

File: master/internal/resourcemanagers/job.go [hunk 2]
```diff
@@ -153,8 +162,11 @@ func computeNewJobPos(
 	return newPos, nil
 }
 
-func initalizeQueuePosition(aTime time.Time) decimal.Decimal {
+func initalizeQueuePosition(aTime time.Time, isK8s bool) decimal.Decimal {
 	// we could add exponent to give us more insertions if needed.
```
[the remark this answers is not in the document]
Comment (eecsliu) [reply master/internal/resourcemanagers/job.go:166]: yeah, it's just a reminder we can bump the value of the exponent if we encounter issues :)

File: master/internal/resourcemanagers/resource_pool.go [hunk 3]
```diff
@@ -470,24 +469,28 @@ func (rp *ResourcePool) moveJob(
 	return nil
 }
 
-func (rp *ResourcePool) findAnchor(
+func findAnchor(
```
[the remark this answers is not in the document]
Comment (eecsliu) [reply master/internal/resourcemanagers/resource_pool.go]: done

**Review by hamidzr — COMMENTED** [review]
this looks good overall. I got into two unexpected cases which could be unrelated to the change here.

- I got into a situation where two jobs were reporting 8/8 slots allocated in the test cluster. the cluster seems to have a max of slots correct?
- an incorrect preemption advertisement? (the API reports jobs as non preemptible yet this happens)
  - submit job 1 
  - submit job 2 => move it above job 1. job 1 is preempted.
  
- can you include the webui patch I shared in your pr please? we can do this separately too.

**Comment by eecsliu** [comment]
I believe the situation where multiple jobs occupy 8 slots is because the first job is still terminating when teh next job gets picked up by the scheduler. It should resolve pretty quickly. The cluster should be running preemption, not sure why it's advertising non-preemption. I'll sync with you on this

**Review by hamidzr — APPROVED** [review]
the changes lgtm. Thanks! 🚀 

I created a ticket for the incorrect preemption reporting
https://determinedai.atlassian.net/browse/DET-7140 that's a separate issue we'll address separately

**Pull request merged**